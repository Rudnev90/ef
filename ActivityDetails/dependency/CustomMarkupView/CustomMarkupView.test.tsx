import * as React from 'react';

import { BcsIconButton } from '@bcs/library';
import { shallow } from 'enzyme';
import { identity } from 'fp-ts/lib/function';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import { CustomMarkupView } from '.';

window.open = jest.fn().mockImplementationOnce(() => {
  return {
    document: {
      write(_message: string) { },
      close() { },
    },
  };
});

const markup = () => <CustomMarkupView message='phone' classes={{}} theme={null}><span /></CustomMarkupView>;
const createJson = () => renderer.create(
  markup(),
  { createNodeMock: identity }, // YB: See https://github.com/facebook/react/issues/7740#issuecomment-247335106
).toJSON();

describe('Section', () => {
  it.concurrent('Should render according to the snapshot', async () => {
    const output: ReactTestRendererJSON = createJson();
    expect(output).toMatchSnapshot();
  });

  it.concurrent('it handle click on icon', async () => {
    const wrapper = shallow(markup());
    wrapper.find(BcsIconButton).simulate('click');
    expect(window.open).toHaveBeenCalled();
  });
});
