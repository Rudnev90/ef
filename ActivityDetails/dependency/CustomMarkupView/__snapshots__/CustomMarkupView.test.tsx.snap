// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-wrap-xs-nowrap-7 MuiGrid-align-items-xs-center-9"
>
  <button
    className="MuiButtonBase-root-137 MuiButton-root-111 BcsButton-root-105 BcsIconButton-root-98 BcsIconButton-smallSizeTheme-100 MuiButton-text-113 MuiButton-textPrimary-114 MuiButton-flat-116 MuiButton-flatPrimary-117"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-112 BcsButton-label-110"
    >
      <span
        className="BcsIconButton-rootLabel-101 BcsIconButton-smallSizeThemeLabel-104"
      >
        <span
          aria-hidden="true"
          className="material-icons MuiIcon-root-140 MuiIcon-fontSizeSmall-147"
        >
          open_in_new
        </span>
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-185"
    />
  </button>
  <h6
    className="MuiTypography-root-149 MuiTypography-subtitle1-167"
  >
    <span />
  </h6>
</div>
`;
